singer/src/main/java/com/pinterest/singer/utils/SingerUtils.java

```diff
@@ -300,4 +298,16 @@ public static String makeGetRequest(String uri) throws IOException {
           throw new IOException(e);
       }
   }
+
+  public static String getHostNameBasedOnConfig(LogStream logStream,
+                                                     SingerConfig singerConfig) {
+    if (singerConfig.isKubernetesEnabled()) {
+      if (logStream.getSingerLog().getPodUid() !=null 
```

Comment:
when will we get `logStream.getSingerLog().getPodUid() == null`?

Reply:
LogStreamManager:L252 since Singer Kube support both local and pods this corner case needs to be covered.